### PR TITLE
TF-1100: Display answered and forwarded

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #1100 

- Resolved:

`Web and desktop`
`Answered and Forwarded`
https://user-images.githubusercontent.com/99852347/230945085-4a1d2a25-da6f-468a-883d-13fc4ad004fa.mov
`Answered`

https://user-images.githubusercontent.com/99852347/231058190-2ebef7e2-42cb-4d94-bd94-4226c8a42291.mov

`Mobile`

[device-2023-04-10-231837.webm](https://user-images.githubusercontent.com/99852347/230946151-78301d76-a535-41a7-b387-08b86088814a.webm)

### CODE REVIEW

**Review by hoangdat — COMMENTED** [review]
I did not see reply icon?

File: lib/features/email/data/network/email_api.dart [hunk 1]
```diff
@@ -162,7 +162,9 @@ class EmailAPI with HandleSetErrorMixin {
             emailRequest.sentMailboxId!.generatePath() : true,
             outboxMailboxId!.generatePath() : null,
             KeyWordIdentifier.emailSeen.generatePath(): true,
-            KeyWordIdentifier.emailDraft.generatePath(): null
+            KeyWordIdentifier.emailDraft.generatePath(): null,
+            KeyWordIdentifier.emailForwarded.generatePath(): true,
+            KeyWordIdentifier.emailAnswered.generatePath(): true,
```
Comment (hoangdat) [inline lib/features/email/data/network/email_api.dart:167]: it should be base on the action of user.

File: lib/features/email/data/network/mdn_api.dart [hunk 2]
```diff
@@ -29,7 +29,9 @@ class MdnAPI {
         request.identityId
     )..addOnSuccessUpdateEmail({
       EmailSubmissionId(ReferenceId(ReferencePrefix.defaultPrefix, request.sendId)): PatchObject({
-        KeyWordIdentifier.mdnSent.toPatchObjectJson(): true
+        KeyWordIdentifier.mdnSent.toPatchObjectJson(): true,
+        KeyWordIdentifier.emailForwarded.toPatchObjectJson(): true,
+        KeyWordIdentifier.emailAnswered.toPatchObjectJson(): true,
```
Comment (hoangdat) [inline lib/features/email/data/network/mdn_api.dart:34]: no need

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart [hunk 3]
```diff
@@ -274,4 +275,46 @@ mixin BaseEmailItemTile {
...
+    if (iconAnsweredOrForwarded(presentationEmail) != null) {
+      return SvgPicture.asset(
+          iconAnsweredOrForwarded(presentationEmail)!,
+          width: width,
+          height: height,
+          colorFilter: AppColor.colorAttachmentIcon.asFilter(),
+          fit: BoxFit.fill);
+    } else {
+      return SizedBox(width: width, height: height);
+    }
```
Comment (hoangdat) [inline lib/features/thread/presentation/mixin/base_email_item_tile.dart:293]: `iconAnsweredOrForwarded` func is called 2 times. Please do it better

File: lib/main/localizations/app_localizations.dart [hunk 4]
```diff
@@ -2833,4 +2833,22 @@ class AppLocalizations {
       'Please input at least one recipient',
       name: 'emptyListEmailForward');
   }
+
+  String get forwardedMessage {
```
Comment (dab246) [inline lib/main/localizations/app_localizations.dart:2837]: Not matched with `name`

File: lib/features/composer/domain/model/email_request.dart [hunk 5]
```diff
@@ -25,6 +28,24 @@ class EmailRequest with EquatableMixin {
...
   ];
+
+  bool get isEmailAnswered => emailActionType == EmailActionType.reply;
+
+  bool get isEmailForwarded => emailActionType == EmailActionType.forward;
+}
+
+extension EmailRequestExtension on EmailRequest {
+
+  EmailRequest toEmailRequest({Email? newEmail}) {
```
Comment (dab246) [inline lib/features/composer/domain/model/email_request.dart:42]: Where is it used?

Comment (nqhhdev) [reply lib/features/composer/domain/model/email_request.dart:42]: I'll delete it

**Review by dab246 — APPROVED** [review]

**Pull request merged**